library/library.ml

```diff
@@ -679,15 +721,20 @@ let error_recursively_dependent_library dir =
 (* Security weakness: file might have been changed on disk between
    writing the content and computing the checksum... *)
 
-let save_library_to ?todo dir f otab =
+(** Note: create_vos could be replaced by a direct read of
+    !Flags.compilation_mode, but this does not seem to be
+    the way things are done for .vio files. *)
+
+let save_library_to ~create_vos ?todo dir f otab =
```

Comment:
I think the API is misleading here. Passing `~todo:(Some ..)` and `~create_vos:true` seems meaningless.
I suggest crafting a data type along these lines
```
type todo_proofs = `NothingToDo (* for vo *) | `Taks of ... Future.t list (* for .vio *) | `EmptyTasks (* for .vos *)
```
It may also make sense to put together the opaque tables and the todo list while you are at it
```
type proofs = { proofs_done : Opaquetables.t ; proofs_todo = [ `Nothing | `Tasks ... | `EmptyTaks ... ] }
```
today the tasks are meant to fill in the missing items in the opaque table.

Reply:
@gares I just tried implementing your suggestion, but there is a catch it seems. At the beginning of `save_library`, I need to compute the variable `except` which lists what proof contents should not be  exported by the global dump. However, with your datatype, I don't have at hand the list of futures of which to extract the identifiers to put into the exclude list. Unless you tell me it is safe to use `let except = Future.UUIDSet.empty` , I would need to  include the list of futures also in the constructor for the `.vio` mode. Please let me know.

Reply:
I have found a way to do this relatively cleanly, by having the constructors for .vos an .vio directly take the list of except as argument.